src/Avalonia.Controls/Menu.cs

```diff
@@ -19,6 +20,36 @@ public class Menu : MenuBase, IMainMenu
 
         private LightDismissOverlayLayer? _overlay;
 
+        /// <summary>
+        /// Defines the <see cref="Opened"/> event.
+        /// </summary>
+        public static readonly RoutedEvent<RoutedEventArgs> OpenedEvent =
```

Comment:
If we have Menu breaking changes, should we follow WPF/UWP APIs then?

In WPF there is no Menu.Opened/Closed event, which makes sense for me (do we really need them?).
But there is [MenuItem.SubmenuOpenedEvent](https://docs.microsoft.com/en-us/dotnet/api/system.windows.controls.menuitem.submenuopenedevent?view=net-5.0) and [MenuItem.SubmenuClosedEvent](https://docs.microsoft.com/en-us/dotnet/api/system.windows.controls.menuitem.submenuclosedevent?view=net-5.0).

In case of UWP I don't see anything like that neither in MenuBar nor MenuBarItem nor MenuFlyoutSubItem.

Reply:
As far as I can tell from #4443, @grokys wanted to simplify and unify naming across controls rather than following WPF. I'm not sure if it's the way we should go with though.

Reply:
Yeah not sure about `Menu.Opened` as the menu itself doesn't really "open" - it's its submenus that open. Having said that `NativeMenu` has an `Opening` event which maps to https://developer.apple.com/documentation/appkit/nsmenudelegate/1518156-menuwillopen?language=objc.